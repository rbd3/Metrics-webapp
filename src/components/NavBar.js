import { FaMicrophone, FaSun } from 'react-icons/fa';

const NavBar = () => (
  <>
    <header className="header">
      <div className="container">
        <ul>
          <li>
            <FaMicrophone />
          </li>
          <li>
            <FaSun />
          </li>
        </ul>
      </div>
    </header>
  </>
);

export default NavBar;
