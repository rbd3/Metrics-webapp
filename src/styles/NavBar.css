.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 2rem;
  background-color: #58163ac9;
  box-shadow: 5px 5px 5px #eae3e3;
}

.logo {
  display: flex;
  gap: 5px;
}

.logo h1 {
  margin-top: 13px;
  color: rgb(245, 233, 233);
}

.header > .logo > img {
  width: 70px;
  height: 70px;
  border-radius: 70px;
}

.header ul {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  list-style-type: none;
  margin-right: 10px;
}

.header ul li {
  color: rgb(243, 204, 216);
  margin-right: 20px;
  font-size: 20px;
}

@media only screen and (max-width: 768px) {
  .header {
    display: flex;
    justify-content: space-between;
    padding: 1.5rem;
  }

  .header > .logo > img {
    width: 50px;
    height: 50px;
    border-radius: 50px;
  }

  .logo h1 {
    margin-top: 6px;
  }
}
