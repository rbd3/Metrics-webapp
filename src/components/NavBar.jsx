import '../styles/NavBar.css';
import { Outlet } from 'react-router-dom';
import { FaMicrophone, FaSun } from 'react-icons/fa';

const NavBar = () => (
  <>
    <header className="header">
      <div className="logo">
        <img src="\assets\movie.jpg" alt="logo" />
        <h1>Movie App</h1>
      </div>
      <div className="container">
        <ul>
          <li>
            <FaMicrophone />
          </li>
          <li>
            <FaSun />
          </li>
        </ul>
      </div>
    </header>
    <Outlet />
  </>
);

export default NavBar;
