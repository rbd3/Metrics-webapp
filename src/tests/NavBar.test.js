// NavBar.test.js

import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NavBar from '../components/NavBar';

describe('NavBar', () => {
  it('should render the NavBar component', () => {
    // Arrange
    const { getByAltText, getByText, getByTestId } = render(<NavBar />);

    // Assert
    expect(getByAltText('logo')).toBeInTheDocument();
    expect(getByText('Movie App')).toBeInTheDocument();
    expect(getByText('Movie App').tagName).toBe('H1');
    expect(getByAltText('logo')).toHaveAttribute('src', '\\assets\\cinema.jpg');

    expect(getByTestId('microphone-icon')).toBeInTheDocument();
    expect(getByTestId('sun-icon')).toBeInTheDocument();
  });
});
