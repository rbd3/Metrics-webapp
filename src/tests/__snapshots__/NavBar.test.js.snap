// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavBar component 1`] = `"<header class=\\"header\\"><div class=\\"logo\\"><img src=\\"\\\\assets\\\\movie.jpg\\" alt=\\"logo\\"><h1>MovieVerse</h1></div><div class=\\"container\\"><ul><li><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 352 512\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z\\"></path></svg></li><li><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 512 512\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z\\"></path></svg></li></ul></div></header>"`;
